### PR TITLE
Fix ifdefs: .NET Core != .NET Standard

### PR DESCRIPTION
**Pull request by knocte**
As discussed in https://github.com/MetacoSA/NBitcoin/pull/444#issuecomment-425695831
we shouldn't be mixing up these ifdefs.

### CODE REVIEW

**Comment by knocte** [comment]
Mmm, only 2 build errors left for this to be green:

```
NBitcoin -> C:\projects\nbitcoin\NBitcoin\bin\Release\net452\NBitcoin.dll
BouncyCastle\util\io\BaseInputStream.cs(42,24): error CS0115: 'BaseInputStream.Close()': no suitable method found to override [C:\projects\nbitcoin\NBitcoin\NBitcoin.csproj]
BouncyCastle\util\io\FilterStream.cs(60,24): error CS0115: 'FilterStream.Close()': no suitable method found to override [C:\projects\nbitcoin\NBitcoin\NBitcoin.csproj]
```

But I don't understand why they happen because the code as it is should be using Dispose instead of Close in the .NET 4.5.2 case. @NicolasDorier do you know why?

**Comment by knocte** [comment]
Nevermind the last message, I fixed the build errors on Windows(AppVeyor). But now I don't know how to read the CI failure in Travis. Any ideas?

**Comment by knocte** [comment]
@NicolasDorier ok, this is ready now! Sorry for the noise before, while I was ironing this out.

File: NBitcoin.Tests/ProtocolTests.cs [hunk 1]
```diff
@@ -860,7 +860,7 @@ public void CanParseReject()
 			Assert.True(reject.Hash == uint256.Parse("964182ffbcec5fafd8f33594b17d6aad4937ff1c59f699e91af44fda94967a57"));
 		}
 
-#if WIN
+#if CLASSICDOTNET
```
Comment (NicolasDorier) [inline NBitcoin.Tests/ProtocolTests.cs]: This test should not be done only on ClassicDotnet

File: NBitcoin.Tests/RPCClientTests.cs [hunk 2]
```diff
@@ -954,7 +954,7 @@ public void CanGetPeersInfo()
 			}
 		}
 #endif
-#if !NOSOCKET && WIN
+#if !NOSOCKET && CLASSICDOTNET
```
Comment (NicolasDorier) [inline NBitcoin.Tests/RPCClientTests.cs]: Can probably remove `CLASSICDOTNET`

**Comment by NicolasDorier** [comment]
I think you can remove all the `PORTABLE` I think it is not used anymore.

**Comment by knocte** [comment]
> I think you can remove all the PORTABLE I think it is not used anymore.

And `SILVERLIGHT`?

**Comment by NicolasDorier** [comment]
yes :D

**Comment by knocte** [comment]
Mmm, if I remove those CLASSICDOTNET, travis doesn't pass. The failure:

```
[39;49m2018-10-02 16:41:37 CreateNewBlock(): block weight: 936 txs: 0 fees: 0 sigops 400
[39;49m[31mError Message:
[39;49m[39;49m[31m Assert.Throws() Failure
Expected: typeof(System.Net.Sockets.SocketException)
Actual:   typeof(System.Net.Internals.SocketExceptionFactory+ExtendedSocketException): No such device or address2018-10-02 16:41:37 UpdateTip: new best=232b0739973dfdef65b9c7bb001f484c14f6f3bfe52ac4cb8b1053d8887e9b55 height=8 version=0x20000000 log2_work=4.169925 tx=9 date='2018-10-02 16:41:40' progress=1.000000 cache=0.0MiB(8txo)

[39;49m2018-10-02 16:41:37 AddToWallet fd71850ff47fede62f4e5fddab7d9a56de0cff2d93570ceed1320b0824e3ff93  new
[39;49m[31m2018-10-02 16:41:37 CreateNewBlock(): block weight: 936 txs: 0 fees: 0 sigops 400
Stack Trace:
[39;49m[39;49m2018-10-02 16:41:37 AddToWallet c599541135530b7516fe9c44eb8bc594b8ae6fb499a35e6ae922d85629f0c216  new
[31m   at System.Net.Dns.InternalGetHostByName(String hostName)
   at System.Net.Dns.GetHostEntry(String hostNameOrAddress)
   at NBitcoin.Utils.ParseIpEndpoint(String endpoint, Int32 defaultPort) in /home/travis/build/MetacoSA/NBitcoin/NBitcoin/Utils.cs:line 851
[39;49m2018-10-02 16:41:37 UpdateTip: new best=678b0c7b6beb2599701b9c77e0ae4ad673567be1585fa630b44b2c033c7b2888 height=9 version=0x20000000 log2_work=4.3219281 tx=10 date='2018-10-02 16:41:40' progress=1.000000 cache=0.0MiB(9txo)
```

**Comment by knocte** [comment]
@NicolasDorier ok, huh, done!

**Comment by NicolasDorier** [comment]
Thanks a lot for the house cleaning!

**Comment by NicolasDorier** [comment]
just pushed new package

**Pull request merged**